### ISSUE TITLE
try to get htm to work with preact-cycle

### ISSUE THREAD

**Issue by weepy** [developit/htm#31]
can't figure out how to do it ^_^

**Comment by Kanaye** [comment]
Hi @weepy!
You just need to import `preact-cycle`s `h`-funciton and `htm.bind()` it.
```js
import {  h } from "preact-cycle";
import htm from "htm";

const html = htm.bind(h);
```

I'ce converted `preact-cycle`s "simple-example" to use `htm` instead of jsx.
You can find that example here: https://codesandbox.io/s/0q15l6no1p

**Comment by weepy** [comment]
I was having problem with importing preact-cycle outside of the jsx
ecosystem.

I.e. You use JSX with the in the index file :
<Contextualize {...props}>

Is it possible to rewrite that without the JSX ?

**Comment by weepy** [comment]
In fact I managed to get that to work, but I'm having trouble converting
the more complex Todo list example. I'm getting this error :

```
preact.js:119 Uncaught DOMException: Failed to execute 'setAttribute' on
'Element': '=' is not a valid attribute name.
    at setAccessor (http://localhost:1234/sandpit/htm/preact.js:119:143)
    at diffAttributes (http://localhost:1234/sandpit/htm/preact.js:285:176)
    at idiff (http://localhost:1234/sandpit/htm/preact.js:204:5)
    at innerDiffNode (http://localhost:1234/sandpit/htm/preact.js:252:15)
    at idiff (http://localhost:1234/sandpit/htm/preact.js:202:61)
    at innerDiffNode (http://localhost:1234/sandpit/htm/preact.js:252:15)
    at idiff (http://localhost:1234/sandpit/htm/preact.js:202:61)
    at diff (http://localhost:1234/sandpit/htm/preact.js:142:15)
    at renderComponent (http://localhost:1234/sandpit/htm/preact.js:401:20)
    at renderComponent (http://localhost:1234/sandpit/htm/preact.js:391:13)
```

It seems to be having trouble with the htm/jsx

Have I made an error :?

```
  const App = ({ text, todos, mutate, mutation }) => (
        html`<div>
          <form onSubmit={mutation(ADD)} action="javascript:">
            <input value={text} onInput={e => mutate('text',
e.target.value)} />
            <button action="submit">Add</button>
          </form>
          <ul>
            ${ todos.map( todo => (
              html`<li onClick={mutation(TOGGLE, todo)}>
                <input type="checkbox" checked={todo.done} readonly />
                <p>{ todo.text }</p>
                <a onClick={mutation(REMOVE, todo)}>✕</a>
              </li>`
            ))}
          </ul>
        </div>`
      );
```

**Comment by marvinhagemeister** [comment]
@weepy you forgot to add the `$`-sign around the braces that is necessary when inserting values into a template string:

```js
const foo = "world";
const str = `Hello ${foo}`;
```

**Comment by developit** [comment]
We should have a section in the README explaining the couple things that need to be modified when going from JSX to HTM.

**Comment by marvinhagemeister** [comment]
Yup, I guess that most users of `htm` are coming from `JSX` so this may make the transition a lot easier 🙂

**Comment by weepy** [comment]
Yeah that’d be useful. Maybe a less cryptic error. Obviously it was my own
foolishness but it’s kinda hard to spot it all as it’s quite a lot of line
noise.
Also it would be worth replacing the Jsx in the index file with ‘h’

J

**Comment by developit** [comment]
This is fixed in 2.0!
